tests/integration/test_analysisd/test_rootcheck/test_configuration/test_rootcheck_delete.py

```diff
@@ -0,0 +1,155 @@
+
+# Test daemons to restart.
+daemons_handler_configuration = {'all_daemons': True}
+
+# Test function.
+@pytest.mark.parametrize('test_configuration, test_metadata',
+                         zip(test_configuration, test_metadata), ids=test_cases_ids)
+def test_rootcheck_delete(test_configuration, test_metadata, set_wazuh_configuration,
+                   daemons_handler, wait_for_rootcheck_start, truncate_monitored_files,
+                   simulate_agents):
+    '''
+    Testing with daemons_handler,
+    description: Check if the 'rootcheck' modules is working properly, that is, by checking if the logs
+                 are deleted correctly.
+                 For this purpose, the test will create a specific number of agents, and enable the rootcheck module.
+                 The rootcheck events will be sent for 60 seconds. After the time has passed, the rootcheck module
+                 gets disabled and check if the logs are deleted from the database when sending the delete
+                 table request.
+
+    wazuh_min_version: 4.2.0
+
+    tier: 0
+
+    parameters:
+        - test_configuration:
+            type: dict
+            brief: Configuration loaded from `configuration_templates`.
+        - test_metadata:
+            type: dict
+            brief: Test case metadata.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Apply changes to the ossec.conf configuration.
+        - wait_for_rootcheck_startup:
+            type: fixture
+            brief: Wait until the 'wazuh-analysisd' has begun and the 'alerts.json' file is created.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Truncate wazuh logs.
+        - daemons_handler:
+            type: fixture
+            brief: Handler of Wazuh daemons.
+        - simulate_agents:
+            type: fixture
+            brief: Handler simulates agents.
+
+    assertions:
+        - Verify that the rootcheck events are deletet from the database
+    input_description: Different test cases are contained in an external YAML file (cases_configuration.yaml)
+                       which includes configuration settings for the 'rootcheck' module.
+    expected_output:
+        - Wazuh DB returned an error trying to delete the agent
+        - Rootcheck events were not deleted
+    '''
+    injectors = []
+    agents = simulate_agents
+
+    for agent in agents:
+        agent.modules['rootcheck']['status'] = 'enabled'
+        _, injector = connect(agent)
+        injectors.append(injector)
+        agents.append(agent)
+
+    # Let rootcheck events to be sent for 60 seconds
+    time.sleep(60)
+
+    for injector in injectors:
+        injector.stop_receive()
+
+    # Service needs to be restarted
+    control_service('start')
+
+    for agent in agents:
+        response = send_request_socket(f'agent {agent.id} rootcheck delete')
```

Comment:
Same here. Create a wrapper in the framework (in `utils/db_queries/agent_db.py`) that uses `query_wdb`, a function that already communicates with `wazuh-db`.
We don't need `send_request_socket` here, but don't remove it from the framework, it can be useful in the future.

Reply:
Query should be something like this:
```
agent {agent_id} rootcheck delete
```
Use the wrapper here, the query should be in the framework.